javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplate.java

```diff
@@ -323,26 +428,24 @@ private static boolean addSpecialHeadersToRequestOptions(JavaBlock function, Cli
             requestOptionsLocal = true;
             function.line("RequestOptions requestOptionsLocal = requestOptions == null ? new RequestOptions() : requestOptions;");
             function.line(String.format("requestOptionsLocal.setHeader(\"%1$s\", UUID.randomUUID().toString());", MethodUtil.REPEATABILITY_REQUEST_ID_HEADER));
-            function.line(String.format("requestOptionsLocal.setHeader(\"%1$s\", DateTimeFormatter.ofPattern(\"EEE, dd MMM yyyy HH:mm:ss z\", Locale.ENGLISH).withZone(ZoneId.of(\"GMT\")).format(OffsetDateTime.now()));", MethodUtil.REPEATABILITY_FIRST_SENT_HEADER));
+            function.line(String.format("requestOptionsLocal.setHeader(\"%1$s\", DateTimeRfc1123.toRfc1123String(OffsetDateTime.now()));", MethodUtil.REPEATABILITY_FIRST_SENT_HEADER));
         }
         return requestOptionsLocal;
     }
 
     protected static void addSpecialHeadersToLocalVariables(JavaBlock function, ClientMethod clientMethod) {
         if (MethodUtil.isMethodIncludeRepeatableRequestHeaders(clientMethod.getProxyMethod())) {
             function.line(String.format("String %1$s = UUID.randomUUID().toString();", MethodUtil.REPEATABILITY_REQUEST_ID_VARIABLE_NAME));
-            function.line(String.format("String %1$s = DateTimeFormatter.ofPattern(\"EEE, dd MMM yyyy HH:mm:ss z\", Locale.ENGLISH).withZone(ZoneId.of(\"GMT\")).format(OffsetDateTime.now());", MethodUtil.REPEATABILITY_FIRST_SENT_VARIABLE_NAME));
+            function.line(String.format("String %1$s = DateTimeRfc1123.toRfc1123String(OffsetDateTime.now());", MethodUtil.REPEATABILITY_FIRST_SENT_VARIABLE_NAME));
```

Comment:
Oh, nice. Is it new in core?

Please add the imports here https://github.com/Azure/autorest.java/blob/7bdee3b94e32822e00a20262fcf3beaa2393c2b4/javagen/src/main/java/com/azure/autorest/model/clientmodel/ClientMethod.java#L422-L427

Reply:
Added here as well. Something we may want to look at in the future is making it easier to add imports to a class file. Right now, imports are only able to be determined by parameters and methods which is good for centralization but then we run into these giant conditional blocks as seen in `ClientMethod` to handle each scenario that would have been a check elsewhere in code. It took me a while to realize this and be able to update the method imports based on the changes I made.

Reply:
Emm, yes, current it is a bit hard to use. For now the simple solution is just to add whatever required import to `ClientMethod` without condition, and let formatter to handle the clean-up.

This is part of the reason I added [MethodTemplate](https://github.com/Azure/autorest.java/blob/main/javagen/src/main/java/com/azure/autorest/template/prototype/MethodTemplate.java) for mgmt to keep related imports and javadoc and codes near, because of the diversity of the type of methods there.
But it got its own drawback. It blurs the line of model and template. So that probably remain in mgmt exclusively.